src/components/button/_button.scss

```diff
@@ -1,80 +1,109 @@
     border-radius: 0;
-    border-color: transparent;
-    outline: 1px solid transparent; // keep some button appearance when changing colour settings in browsers
-    outline-offset: -1px; // fixes bug in Safari that outline width on focus is not overwritten, is reset to 0 on focus in govuk_template
+    color: $govuk-button-text-colour;
     background-color: $govuk-button-colour;
-    box-shadow: 0 2px 0 $govuk-button-colour-darken-15;
-    font-family: $govuk-font-stack;
-    // Had to use important because th linter won't let me put this under the font-size include
-    line-height: 1 !important;
+    box-shadow: 0 $button-shadow-size 0 $button-shadow-colour; // s0
     text-align: center;
     text-decoration: none;
     vertical-align: top;
+
     cursor: pointer;
     -webkit-appearance: none;
 
     @include mq($from: tablet) {
       width: auto;
     }
 
-    // Set text colour depending on background colour
-    @if lightness($govuk-button-colour) < 50% {
-      color: $govuk-white;
-
-      &:link,
-      &:hover,
-      &:focus,
-      &:visited {
-        color: $govuk-white;
-      }
-    } @else {
-      color: $govuk-text-colour;
-
-      &:link,
-      &:hover,
-      &:focus,
-      &:visited {
-        color: $govuk-text-colour;
+    @include ie-lte(8) {
+      // Substitute shadow for border in IE8
+      padding: ($govuk-spacing-scale-2 - $govuk-border-width-form-element) $govuk-spacing-scale-2;
+      border-bottom: $button-shadow-size solid $button-shadow-colour; // s0
+
+      // Fixes a bug where IE8 puts a black border around certain elements
+      // sass-lint:disable no-color-literals
+      &[type="submit"],
+      &[type="reset"],
+      &[type="button"] {
+        filter: chroma(color = #000000);
       }
+      // sass-line:enable no-color-literals
     }
 
+    // Fix unwanted button padding in Firefox
+    &:-moz-focus-inner {
+      padding: 0;
+      border: 0;
+    }
+
+    &:link,
+    &:hover,
+    &:focus,
     &:visited {
-      background-color: $govuk-button-colour;
+      color: $govuk-button-text-colour;
     }
 
     &:hover,
     &:focus {
       background-color: darken($govuk-button-colour, 5%);
     }
 
+    &:focus {
+      outline: $govuk-focus-width solid $govuk-focus-colour;
+    }
+
+    // 'Depress' the button when clicked by removing the shadow and shifting
+    // the button down by the shadow width
     &:active {
-      top: $govuk-border-width-form-element;
-      box-shadow: 0 0 0 $govuk-button-colour;
+      top: $button-shadow-size;
+      box-shadow: none;
     }
 
-    // Fix unwanted button padding in Firefox
-    &:-moz-focus-inner {
-      padding: 0;
-      border: 0;
+    // These adjustments do not work for <input type="button"> as non-container
+    // elements cannot include pseudo elements.
+
+    // Expand the click target to cover the shadow as well, in case users try
+    // to click it.
+    &::before {
```

Comment:
Does this account for the `box-shadow`? In the original there's a `110%` definition since if you try and click the `box-shadow` it does not work.